### ISSUE TITLE
MTU configuration

### ISSUE THREAD

**Issue by n0k0m3** [netbirdio/netbird#743]
**Is your feature request related to a problem? Please describe.**
I want to be able to change MTU (similar to vanilla Wireguard or Netmaker), either through local configuration or management UI

**Describe the solution you'd like**
Expose MTU configuration to /etc/netbird/config.json or management UI, and config will either be loaded with `-c` or update when `up` is called 

**Describe alternatives you've considered**
None, as NetBird doesn't expose MTU configuration to client

**Additional context**
None

**Comment by ThHirsch** [comment]
I am looking for a solution/fix as well.
We were not able to get Netbird working on secure connections (https, ssh, ...) after (initial) successfull handshake with the servicec to the target client, if the source client is in a networt where the router is connected to a DOCIS3.1 cable modem (it seems it adds/needs additional encapsulation then).
Especially if the connection is then going to send the data in bigger packets, which then get fragmented because of the MTU and breaking the secure connection checks then.  (at typical MTU Problem?!).
Fiddling with the MTU on other levels of the connection did not help - so it seems Netbirds client MTU must be configurable. BTW - I am on MAC OSX 14.2.1
So please consider to include this in the published client, as I am not a delvelooer and I am not able to compile own client with the fixes/patches that seem to be provided by others.

**Comment by maisen20** [comment]
@ThHirsch have you found a workaround to fix this problem? It seems that i have the same problem as you, a network of mine is also connected to the internet via a DOCIS3.1 cable modem.

I think this is why all pings between the networks work without problems (from A to B and B to A), but I can only access all WebUIs and SSH servers in one way. So from network B to A everything works fine, but from network A to B I can only ping all clients in the remote network and nothing more (no WebUI access and no SSH connections of/to clients in Network B).

**Comment by ThHirsch** [comment]
@maisen20 :  No, we ended up in configuring the (lower) MTU on the (WLAN-)Network-Card in the OS (MacOSX in my case).
It seems, if it is configured there, Wireguard/Netbird will 'magically' honor this then (or at least adapt to it).
Hopt this helps for your situation.

**Comment by alexcupertme** [comment]
Is there any updates on this problem? We also have troubles with inability to change MTU while connecting hosts with SSH

**Comment by ThHirsch** [comment]
As tis is still bugging me in our environments, I am keen to see this pcked up and solved.
Especially, as  OpenDUT seemed to have contributed some proposals and even code for a solution?
Is there any progres?
Is it on any roadmap?
Sorry - as I am not a developer, I just can't help in making it happen other than by beeing sitcky... ;-)

**Comment by PapaZigE** [comment]
+1 On this. I have been scratching my head for weeks on what my network issue was and changing the MTU for the Netbird interface has solved it. 

Netbird server runs on my VPS and meshes with an on-prem server that runs it's own services exposed via Netbird. HTTP requests were working as expected however, SMTP was not for ALL containers and I could not figure out why for the life of me. Finally, I stumbled upon this link [Network Issue Timeout](https://stackoverflow.com/questions/76915016/network-issue-timeout-in-docker-vs-working-in-host-tls-issue-leads-to-not-rea), updated the MTU for my servers Netbird interfaces and now, things are working again. It would be awesome to have this in a config so it persists.

**Comment by nazarewk** [comment]
Hello @n0k0m3,

We're currently reviewing our open issues and would like to verify if this problem still exists in the [latest NetBird version](https://github.com/netbirdio/netbird/releases).

Could you please confirm if the issue is still there?

We may close this issue temporarily if we don't hear back from you within **2 weeks**, but feel free to reopen it with updated information.

Thanks for your contribution to improving the project!

**Comment by qay21** [comment]
Hi @nazarewk 
I don't want to speak for @n0k0m3 but I'm currently investigating a very low bandwidth issue with Netbird (<30Mbps whereas pure Wireguard happily download at 600+Mbps on a 1Gpbs LAN). I'm quite uncertain of the cause at the moment, but I stumbled upon this Github issue precisely because I was wondering why netbird's interface's MTU is so low (1280). It still doesn't seem to be configurable, or at least I don't see how and can't find any documentation on the matter.

**Comment by alexcupertme** [comment]
Yes, the problem persists

**Comment by Alexj12** [comment]
+1, would be very handy to have this.

**Comment by nazarewk** [comment]
FYI: https://github.com/netbirdio/netbird/pull/4213

**Comment by Alexj12** [comment]
I see the related PR has now been merged in and released. Apologies if I've missed something, but I have searched for documentation changes and haven't been able to track down the relevant information on how to set this centrally, as I note the below line. Is this supported on a self hosted server?

> Adds a mtu field to the mgmt proto to set this from a central location

**Comment by jammin84** [comment]
Now that there is a "Netbird up --mtu" setting deployed ... is there roadmap to setting this globally from the management side either globally or to a group?

**Comment by TechHutTV** [comment]
Configurable MTU landed in PR #4213 and shipped in v0.56.0 — `netbird up --mtu <value>` now lets you set the WireGuard interface MTU per-peer, and the management proto carries an `mtu` field for central configuration. Closing as fixed; please reopen if you still see MTU-related issues on the latest release. 

@jammin84 we are cleaning up old issues, please open a discussion with any feature requests.
@Alexj12 it's listed here in the [CLI docs](https://docs.netbird.io/get-started/cli). 

If I missed anything feel free to reply here or open a new discussion.
